ibm/resource_ibm_is_instance.go

```diff
@@ -996,7 +996,12 @@ func isWaitForInstanceAvailable(instanceC *vpcv1.VpcV1, id string, timeout time.
 		forceTimeout := v.(int)
 		go isRestartStartAction(instanceC, id, d, forceTimeout, communicator)
 	}
-	return stateConf.WaitForState()
+	ins, err := stateConf.WaitForState()
+	instance := ins.(*vpcv1.Instance)
```

Comment:
Can we add this condition in 
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/95d3c930382526e8920aed8e483e7e0ed46af471/ibm/resource_ibm_is_instance.go#L1050

```
if *instance.Status == "failed" ] {
			// let know the isRestartStartAction() to stop
			close(communicator)
			return instance, *instance.Status, fmt.Errorf("XXXXX")

		}
		```